javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithExpression.java

```diff
@@ -0,0 +1,10 @@
+package com.github.javaparser.ast.nodeTypes;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.expr.Expression;
+
+public interface NodeWithExpression<N extends Node> {
```

Comment:
They are nodes containing an expression. They can be expressions themselves. The interface is really general as you can see, but it is easy for refactoring ;)